inbm/dispatcher-agent/dispatcher/source/source_command.py

```diff
@@ -113,16 +112,29 @@ def _handle_app_source_command(
         return Result(status=200, message=serialized_list)
 
     if "remove" in app_action:
-        keyname = parsed_head.get_children("applicationSource/remove/gpg")["keyname"]
+        keyname = None
+        try:
+            keyname = parsed_head.get_children("applicationSource/remove/gpg")["keyname"]
+        except XmlException:
+            # These children may not be present
+            logger.info(f"Optional GPG key parameters not present in manifest")
         filename = parsed_head.get_children("applicationSource/remove/repo")["filename"]
         application_source_manager.remove(
-            ApplicationRemoveSourceParameters(file_name=filename, gpg_key_name=keyname)
+            ApplicationRemoveSourceParameters(source_list_file_name=filename, gpg_key_name=keyname)
         )
```

Comment:
**Maintainability**
**Suggestion:** Refactor the repeated try-except blocks for GPG key extraction into a separate function to improve code maintainability and reduce duplication.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        keyname = extract_gpg_keyname(parsed_head, "applicationSource/remove/gpg")
        filename = parsed_head.get_children("applicationSource/remove/repo")["filename"]
        application_source_manager.remove(
            ApplicationRemoveSourceParameters(source_list_file_name=filename, gpg_key_name=keyname)
        )
        return Result(status=200, message="SUCCESS")

        def extract_gpg_keyname(parsed_head: XmlHandler, path: str) -> Optional[str]:
            try:
                return parsed_head.get_children(path)["keyname"]
            except XmlException:
                logger.info(f"Optional GPG key parameters not present in manifest")
                return None
```